src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ConversationDTOService.java

```diff
@@ -0,0 +1,208 @@
+        var oneToOneChatDTO = new OneToOneChatDTO(oneToOneChat);
+        oneToOneChatDTO.setIsMember(participantOfRequestingUser.isPresent());
+        participantOfRequestingUser.ifPresent(conversationParticipant -> oneToOneChatDTO.setLastReadDate(conversationParticipant.getLastRead()));
+        participantOfRequestingUser.ifPresent(conversationParticipant -> oneToOneChatDTO.setUnreadMessagesCount(conversationParticipant.getUnreadMessagesCount()));
+        participantOfRequestingUser.ifPresent(conversationParticipant -> oneToOneChatDTO.setIsFavorite(conversationParticipant.getIsFavorite()));
+        participantOfRequestingUser.ifPresent(conversationParticipant -> oneToOneChatDTO.setIsHidden(conversationParticipant.getIsHidden()));
+        oneToOneChatDTO.setMembers(chatParticipants);
+        if (oneToOneChat.getCreator() != null) {
+            oneToOneChatDTO.setIsCreator(oneToOneChat.getCreator().getId().equals(requestingUser.getId()));
+        }
+        else {
+            log.warn("Unexpected Behaviour: OneToOneChat {} has no creator. Can happen with db entries before December 2022", oneToOneChat.getId());
+            oneToOneChatDTO.setIsCreator(false);
+        }
+        oneToOneChatDTO.setNumberOfMembers(conversationParticipants.size());
+        return oneToOneChatDTO;
+    }
+
+    /**
+     * Creates a GroupChatDTO from a GroupChat
+     *
+     * @param requestingUser the user requesting the DTO
+     * @param groupChat      the group chat to create the DTO from
+     * @return the created GroupChatDTO
+     */
+    @NotNull
+    public GroupChatDTO convertGroupChatToDto(User requestingUser, GroupChat groupChat) {
+        var course = groupChat.getCourse();
+        Set<ConversationParticipant> conversationParticipants = getConversationParticipants(groupChat);
+        var participantOfRequestingUser = conversationParticipants.stream()
+                .filter(conversationParticipant -> conversationParticipant.getUser().getId().equals(requestingUser.getId())).findFirst();
+        Set<ConversationUserDTO> chatParticipants = getChatParticipantDTOs(requestingUser, course, conversationParticipants);
+        var groupChatDTO = new GroupChatDTO(groupChat);
+        groupChatDTO.setIsMember(participantOfRequestingUser.isPresent());
+        participantOfRequestingUser.ifPresent(conversationParticipant -> groupChatDTO.setLastReadDate(conversationParticipant.getLastRead()));
+        participantOfRequestingUser.ifPresent(conversationParticipant -> groupChatDTO.setUnreadMessagesCount(conversationParticipant.getUnreadMessagesCount()));
+        participantOfRequestingUser.ifPresent(conversationParticipant -> groupChatDTO.setIsFavorite(conversationParticipant.getIsFavorite()));
+        participantOfRequestingUser.ifPresent(conversationParticipant -> groupChatDTO.setIsHidden(conversationParticipant.getIsHidden()));
```

Comment:
Same as above.

Reply:
Done in 11008eb152db6fbddc6487737ac0b019b0b742b7